core/rawdb/chain_freezer.go

```diff
@@ -282,6 +276,18 @@ func (f *chainFreezer) freeze(db ethdb.KeyValueStore) {
 				last = first + freezerBatchLimit
 			}
 		}
+
+		// check env first before chain freeze, it must wait when the env is necessary
+		if err := f.checkFreezerEnv(); err != nil {
+			f.waitEnvTimes++
+			if f.waitEnvTimes >= maxWaitFreezerEnvTimes {
```

Comment:
agree to move it after the previous threshold check.
but better not add this try limit, in case it could take more than 3min to make it ready.